MarvellOS.py

```diff
@@ -656,9 +682,13 @@ def sqroot(self):
             except OverflowError as e:
                 self.ans["text"]="Out of range"
 
-        def fact(self):
+        def fact(self,event=None):
             try:
-                self.a=self.num.get()
+                self.v=self.num.get()
+                if "!" in self.v:
+                   self.a=self.v[:-1]
+                else:
+                    self.a=self.v
                 if float(self.a)>=0:
                     self.an=str(math.factorial(float(self.a)))
```

Comment:
A bit unrelated, but can you please change the `float()` here to `int()`. I noticed that the factorial function has been changed to take only int values and raises an error currently, so it'll be good to fix it in this PR itself since it deals with the calculator.